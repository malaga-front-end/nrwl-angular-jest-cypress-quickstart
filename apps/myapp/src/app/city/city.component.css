h1 {
    font: 300 40px Helvetica, Verdana, sans-serif;
    position: fixed;
    top: 20px;
    right: 0;
    background: hsla(80, 90%, 40%, 0.7);
    border: solid hsla(80, 90%, 40%, 0.5);
    color: white;
    text-align: right;
    border-right: none;
    padding: 0.5em 0.5em 0.5em 1em;
    box-shadow: 0 1px 3px black;
    border-radius: 3em 0 0 3em;
    transition: bottom 0.5s ease-in-out;
    animation-name: snackbar-show;
    animation-duration: 0.5s;
  }
  
  @keyframes snackbar-show {
    0%{ right: -200px; }
  }
  