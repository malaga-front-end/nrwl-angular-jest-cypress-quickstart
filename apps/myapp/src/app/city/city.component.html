<h1 *ngIf="getCity() | async as city">{{ city }}</h1>