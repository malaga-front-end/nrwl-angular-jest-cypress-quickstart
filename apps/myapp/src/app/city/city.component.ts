import { Component, OnInit } from '@angular/core';
import { SharedService } from '../shared.service';

@Component({
  selector: 'myapp-city',
  templateUrl: './city.component.html',
  styleUrls: ['./city.component.css']
})
export class CityComponent implements OnInit {

  constructor(private sharedService: SharedService) { }

  ngOnInit(): void {
  }

  getCity() {
    return this.sharedService.getCity();
  }

}
