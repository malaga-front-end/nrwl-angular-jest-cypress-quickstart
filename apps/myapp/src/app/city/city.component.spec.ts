import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';

import { CityComponent } from './city.component';
import { SharedService } from '../shared.service';
import { of } from 'rxjs';

jest.mock('../shared.service');

describe('CityComponent', () => {
  let component: CityComponent;
  let fixture: ComponentFixture<CityComponent>;
  let sharedService: SharedService;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [ CityComponent ],
      providers: [ SharedService ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    sharedService = TestBed.inject(SharedService);
    jest.spyOn(sharedService, 'getCity').mockReturnValue(of('Madrid'));
    fixture = TestBed.createComponent(CityComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render a city', () => {
    expect(fixture).toMatchSnapshot();
  });
});
