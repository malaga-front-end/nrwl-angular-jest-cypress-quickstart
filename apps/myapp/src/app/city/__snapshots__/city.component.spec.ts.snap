// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityComponent should render a city 1`] = `
<myapp-city
  __ngContext__={[Function LRootView]}
  sharedService={[Function SharedService]}
>
  <h1>
    Madrid
  </h1>
</myapp-city>
`;
